samples/azure-functions/service-bus/ASBFunctions_2/AzureFunctions.ASBTrigger.FunctionsHostBuilder/AzureFunctions.ASBTrigger.FunctionsHostBuilder.csproj

```diff
@@ -1,12 +1,12 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
-    <AzureFunctionsVersion>v3</AzureFunctionsVersion>
+    <TargetFramework>net6.0</TargetFramework>
+    <AzureFunctionsVersion>v4</AzureFunctionsVersion>
```

Comment:
@jenspettersson until we fix this I have verified that you can work around this by implementing your own trigger class like this: (you need to comment out `[assembly:NServiceBusTriggerFunction("ASBWorkerEndpoint")]` to avoid the trigger to be autogenerated)

```
public class FunctionEndpointTrigger
{
    IFunctionEndpoint endpoint;

    public FunctionEndpointTrigger(IFunctionEndpoint endpoint)
    {
        this.endpoint = endpoint;
    }


    [Function("NServiceBusFunctionEndpointTrigger-ASBWorkerEndpoint")]
    public async Task Run(
        [ServiceBusTrigger("ASBWorkerEndpoint")] byte[] messageBody,
        IDictionary<string, object> userProperties,
        string messageId,
        int deliveryCount,
        string replyTo,
        string correlationId,
        FunctionContext context)
    {
        await endpoint.Process(messageBody, userProperties.ToDictionary(k => k.Key, k => k.Value.ToString()), messageId, deliveryCount, replyTo, correlationId, context);
    }
}

```

Reply:
Note that the change is to use `IDictionary<string, object> userProperties,` instead of `IDictionary<string, string> userProperties,`

Reply:
Thanks, I'll take a look when I've got time. Currently occupied elsewhere. Btw, I guess this is the workaround for the second issue, any news about the first with the container being disposed when running on the v4 host InProcess?